File: src/test/java/com/jiro4989/tkfm/util/ImageUtilTest.java
package com.jiro4989.tkfm.util;

import static org.junit.jupiter.api.Assertions.*;

import java.io.File;
import java.io.IOException;
import javafx.scene.image.Image;
import org.junit.jupiter.api.*;
import org.junit.jupiter.api.extension.ExtendWith;
import org.testfx.framework.junit5.*;

@ExtendWith(ApplicationExtension.class)
public class ImageUtilTest {
  @BeforeEach
  public void beforeEach() {
    File dir = new File("tmp");
    dir.mkdirs();
  }

  @AfterEach
  public void afterEach() {
    File dir = new File("tmp");
    for (var f : dir.listFiles()) f.delete();
    dir.delete();
  }

  @Test
  public void testWriteFile() throws IOException {
    var image = new Image("20x20.png");
    var path = "tmp" + File.separator + "out.png";
    var file = new File(path);
    assertFalse(file.exists());
    ImageUtil.writeFile(image, file);

    file = new File(path);
    assertTrue(file.exists());
  }
}

File: src/test/kotlin/com/jiro4989/tkfm/util/ImageUtilTest.kt
package com.jiro4989.tkfm.util

import java.io.File
import java.io.IOException
import javafx.scene.image.Image
import org.junit.jupiter.api.*
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.extension.ExtendWith
import org.testfx.framework.junit5.*

@ExtendWith(ApplicationExtension::class)
class ImageUtilTest {
  @BeforeEach
  fun beforeEach() {
    val dir = File("tmp")
    dir.mkdirs()
  }

  @AfterEach
  fun afterEach() {
    val dir = File("tmp")
    dir.listFiles().forEach { it.delete() }
    dir.delete()
  }

  @Test
  fun testWriteFile() {
    val image = Image("20x20.png")
    val path = "tmp" + File.separator + "out.png"
    val file = File(path)
    assertFalse(file.exists())
    writeFile(image, file)
    assertTrue(File(path).exists())
  }
}
